lib/puppet/type/cisco_route_map.rb

```diff
@@ -0,0 +1,1642 @@
+          Valid values are integer, keyword 'default'."
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
+  end # property set_distance_local
+
+  newproperty(:set_extcomm_list) do
+    desc "Set BGP extended community list (for deletion). Valid values are string,
+          keyword 'default'"
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property set_extcomm_list
+
+  newproperty(:set_extcommunity_4bytes_additive) do
+    desc 'Add to existing generic extcommunity'
+
+    newvalues(:true, :false, :default)
+  end # property set_extcommunity_4bytes_additive
+
+  newproperty(:set_extcommunity_4bytes_non_transitive, array_matching: :all) do
+    format = '[nt1, nt2]'
+    desc 'Set non-transitive extended community. An array of [nt1, nt2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property set_extcommunity_4bytes_non_transitive
+
+  newproperty(:set_extcommunity_4bytes_none) do
+    desc 'Set no extcommunity generic attribute'
+
+    newvalues(:true, :false, :default)
+  end # property set_extcommunity_4bytes_none
+
+  newproperty(:set_extcommunity_4bytes_transitive, array_matching: :all) do
+    format = '[tr1, tr2]'
+    desc 'Set transitive extended community. An array of [tr1, tr2 and so on]' \
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property set_extcommunity_4bytes_transitive
+
+  newproperty(:set_extcommunity_cost_igp, array_matching: :all) do
+    format = '[[communityId, cost], [cid, co]]'
+    desc 'An array of [communityId, cost] pairs. '\
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
```

Comment:
Not sure I understand why you are overriding the `insync?` method in this way.  Let's chat more about this.